terralab/utils.py

```diff
@@ -29,29 +29,78 @@ def wrapper(*args, **kwargs):
-            LOGGER.error(add_blankline_after(formatted_message))
-            LOGGER.error(add_blankline_after(SUPPORT_EMAIL_TEXT))
+            LOGGER.error(add_blankline_before(formatted_message))
+            LOGGER.error(add_blankline_before(SUPPORT_EMAIL_TEXT))
             exit(1)
         except Exception as e:
-            LOGGER.error(add_blankline_after(str(e)))
-            LOGGER.error(add_blankline_after(SUPPORT_EMAIL_TEXT))
+            LOGGER.error(add_blankline_before(str(e)))
+            LOGGER.error(add_blankline_before(SUPPORT_EMAIL_TEXT))
             exit(1)
 
     return wrapper
 
 
-def process_json_to_dict(json_data) -> dict:
-    """Process the given JSON string and return a dictionary.
-    Returns None if the input string is not able to be parsed to a dictionary."""
-    try:
-        data = json.loads(json_data)
-        if not (isinstance(data, dict)):
-            raise TypeError
-        LOGGER.debug(f"Processed inputs: {data}")
-        return data
-    except (json.JSONDecodeError, TypeError):
-        LOGGER.error(add_blankline_after("Input string not parsable to a dictionary."))
-        exit(1)
+def process_inputs_to_dict(inputs: tuple) -> dict:
+    """Process command line arguments (input as a tuple) into a dictionary.
+
+    Handles arguments in the format:
+    - '--key value'
+    - '--key=value'
+    - '--flag' (value will be None)
+
+    Args:
+        inputs: Tuple of string arguments, generated by Click
+
+    Returns:
+        Dictionary of processed key-value pairs
+
+    Raises:
+        ValueError: If arguments are improperly formatted or contain duplicate keys
+    """
+    inputs_dict = {}
+    i = 0
+
+    while i < len(inputs):
+        current_arg = inputs[i]
+
+        if not current_arg.startswith("--"):
+            raise ValueError(
+                f"Error: Invalid input key '{current_arg}'. All input keys must start with '--'."
+            )
+
+        # strip leading dashes
+        arg_without_dashes = current_arg[2:]
+
+        if "=" in arg_without_dashes:
+            # Handle --key=value format
+            key, value = arg_without_dashes.split("=", 1)
+            value = process_value(value)
+        else:
+            # Handle --key value or --flag format
+            key = arg_without_dashes
+            if i + 1 < len(inputs) and not inputs[i + 1].startswith("--"):
+                value = process_value(inputs[i + 1])
+                i += 1
+            else:
+                value = None
+
+        if key in inputs_dict:
+            raise ValueError(f"Error: Duplicate input key found: '{key}'.")
+
+        LOGGER.debug(f"Processed input: {key}={value}")
+        inputs_dict[key] = value
+        i += 1
+
+    return inputs_dict
+
+
+def process_value(raw_value: str):
+    """Process a raw input value string, splitting to an array if commas are present."""
+    # process arrays
+    if "," in raw_value:
+        return raw_value.split(",")
```

Comment:
it seems like we're allowing for array style inputs, do we need to any checks to validate_pipeline_inputs?  like for basename do we now need to check the value in the dict isnt an array?

Reply:
we can just rely on the service to do that kind of check